lib/pbench/test/unit/server/query_apis/conftest.py

```diff
@@ -60,3 +60,53 @@ def query_api(
         return response
 
     return query_api
+
+
+@pytest.fixture
+@responses.activate
+def query_api_get(client, server_config, provide_metadata):
+    """
+    Help controller queries that want to interact with a mocked
+    Elasticsearch service.
+
+    This is a fixture which returns a function that can be
+    used to set up and validate a mocked Elasticsearch query with a JSON
+    payload and an expected status.
+
+    Parameters to the mocked Elasticsearch GET are passed as keyword
+    parameters: these can be any of the parameters supported by the
+    responses mock. Exceptions are specified by providing an Exception()
+    instance as the 'body'.
+
+    :return: the response object for further checking
+    """
+
+    def query_api(
+        pbench_uri: str,
+        es_uri: str,
+        expected_index: str,
+        expected_status: str,
+        headers: dict = {},
+        **kwargs,
+    ) -> requests.Response:
+        host = server_config.get("elasticsearch", "host")
+        port = server_config.get("elasticsearch", "port")
+        es_url = f"http://{host}:{port}{expected_index}{es_uri}"
+        with responses.RequestsMock() as rsp:
+            """
+            This set up mocks the Server's call to Elasticsearch,
+            unless when the expected status is either FORBIDDEN or UNAUTHORIZED
+            """
+            if expected_status not in [
+                HTTPStatus.FORBIDDEN,
+                HTTPStatus.UNAUTHORIZED,
+            ]:
+                rsp.add(responses.GET, es_url, **kwargs)
+            response = client.get(
```

Comment:
Why are we hiding the actual test in a fixture?

Reply:
Technically this isn't a fixture; it's a customized helper method created using the magic of fixtures. The pattern simplifies a lot of test cases by pushing common fixture dependencies (e.g., `client, server_config, provide_metadata`) out of the individual test case declarations and returns the constructed `query_api` to be called where needed.

Of course it could have been done other ways, but this pattern goes way back to the start of our Flask "unit" tests and it wasn't invented here.